themes/base.theme.bash

```diff
@@ -171,6 +173,15 @@ function git_prompt_minimal_info {
 }
 
 function git_prompt_vars {
+  if ${SCM_GIT_USE_GITSTATUSD}; then # use faster gitstatusd
```

Comment:
It's a good idea to fall back to the regular workflow if gitstatusd fails. You can also avoid calling `git` unnecessarily if gitstatusd succeeds and says the current directory is not a Git repo.

```bash
if ${SCM_GIT_USE_GITSTATUSD} && gitstatus_query; then
  if [[ "${VCS_STATUS_RESULT}" == "ok-sync" ]]; then
    echo "the current directory is a Git repo; VCS_STATUS_* parameters are set; no need to call 'git'"
  else
    echo "the current directory is not a Git repo; VCS_STATUS_* parameters are not set; no need to call 'git'"
  fi
else
  echo "gitstatusd either disabled or doesn't work; must use 'git' commands"
fi
```

Reply:
I think the current variant is ok, because if you look at the context in which prompt_vars function is called, it only happens if it already has been detected that a git repo is present.

Reply:
OK. That initial check for being in a Git repo could be avoided but I guess such optimizations can be done later. There is also a race between the check and `gitstatus_query` but it's unlikely to matter.

As for this code, it might still be a good idea to change it a bit. The docs say to check for exit code of `gitstatus_query` and to look at `VCS_STATUS_RESULT`. By doing what the docs recommend you'll stay away from accidental changes and bugs in gitstatusd. It's also a good idea to have just two possible values for `SCM_GIT_GITSTATUSD_RAN` (`true` and `false`) rather than three (`true`, `false` and unset).

Perhaps something like this?

```bash
if ${SCM_GIT_USE_GITSTATUSD} && gitstatus_query && [[ "${VCS_STATUS_RESULT}" == "ok-sync" ]]; then
  # gitstatusd has successfully retrieved Git status; VCS_STATUS_* parameters are set
  SCM_GIT_GITSTATUSD_RAN=true
else
  # either gitstatusd is disabled, it failed, or the current directory is not a Git repo
  SCM_GIT_GITSTATUSD_RAN=false
fi
```

Reply:
I like that. Will also add a default init for the RAN variable.
Although it shouldn't make a difference because I always checked it in a quoted manner, meaning even if it wouldn't exist the check does work. But better safe than sorry.

Reply:
ok, pushed an update to resolve this.
Also realized the initial scm git call will not happen as this is done:
` elif [[ -f .git/HEAD ]] && [[ -x "$GIT_EXE" ]]; then SCM=$SCM_GIT`

This is a call to check the presence of HEAD, this should be very quick.
So the other call is only used in case of fallback.